spec.bs

```diff
@@ -18,3 +18,50 @@ Default Biblio Status: current
 Markup Shorthands: markdown yes
 Assume Explicit For: yes
 </pre>
+
+
+<h2 id="advertisements">Advertisements</h2>
+
+An <dfn export>advertisement</dfn>
+is a [=struct=] with the following [=struct/items=]:
+
+* An <dfn for=advertisement>render URL</dfn>. The render URL must have “https” as the scheme.
+* An optional <dfn for=advertisement>metadata</dfn> which is text containing opaque JSON encoded data, passed as an object to auction worklet.
+
+
+<h2 id="advertisement-interest-groups">Advertisement Interest Groups</h2>
+
+An <dfn export>advertisement interest group</dfn>
+is a [=struct=] with the following [=struct/items=]:
+ 
+* An <dfn for="advertisement interest group">expiration</dfn> date.   base::Time expiry;
+
+* An <dfn for="advertisement interest group">owner origin</dfn>, an [=origin=] url.
+
+* A <dfn for="advertisement interest group">name</dfn>, a [=string=].
+ 
+* A <dfn for="advertisement interest group">priority</dfn>, a double, initiallized to 0.
+ 
+* An <dfn for="advertisement interest group">enable_bidding_signals_prioritization</dfn>, a boolean, initialized to false.
+ 
+* An optional <dfn for="advertisement interest group">priority_vector</dfn>, A flat [=map=] `<base::flat_map<std::string, double>>`
+ 
+* An optional <dfn for="advertisement interest group">priority_signals_overrides</dfn>, A flat [=map=] `<base::flat_map<std::string, double>>`

```

Comment:
<base::flat_map<std::string, double>> is a C++ thing; we should probably turn this into spec English.  I think specs use this map definition: https://infra.spec.whatwg.org/#maps

Reply:
I wasn't sure what a flat_map was, and whether the difference with map or ordered map was significant.

Reply:
Changed to "flat [=map=] of {{double}}".  Maybe we can (should?) delete the "flat" part.

Reply:
The "flat" part is not relevant for the "infra" version of the code.